examples/cpp-tutorial-4-gles3-pointcloud.cpp

```diff
@@ -0,0 +1,769 @@
+
+                  uniform float u_depth_scale_in_meter;
+                  uniform mat4 u_depth_to_color;
+
+                  uniform vec2 u_color_offset;
+                  uniform vec2 u_color_focal_length;
+                  uniform float u_color_coeffs[5];
+                  uniform int u_color_model;
+
+                  uniform vec2 u_depth_offset;
+                  uniform vec2 u_depth_focal_length;
+                  uniform float u_depth_coeffs[5];
+                  uniform int u_depth_model;
+
+                  out vec2 v_tex;
+
+                  vec3 depth_deproject(vec2 pixel, float depth)
+                  {
+                      vec2 point = (pixel - u_depth_offset) / u_depth_focal_length;
+                      if(u_depth_model == RS_DISTORTION_INVERSE_BROWN_CONRADY)
+                      {
+                          float r2 = dot(point, point);
+                          float f = 1.0 + u_depth_coeffs[0] * r2 + u_depth_coeffs[1] * r2 * r2 +
+                                    u_depth_coeffs[4] * r2 * r2 * r2;
+                          float ux = point.x * f + 2.0 * u_depth_coeffs[2] * point.x * point.y +
+                                     u_depth_coeffs[3] * (r2 + 2.0 * point.x * point.x);
+                          float uy = point.y * f + 2.0 * u_depth_coeffs[3] * point.x * point.y +
+                                     u_depth_coeffs[2] * (r2 + 2.0 * point.y * point.y);
+                          point = vec2(ux, uy);
+                      }
+                      return vec3(point * depth, depth);
+                  }
+
+                  vec2 color_project(vec3 point)
+                  {
+                      vec2 pixel = point.xy / point.z;
+                      if(u_color_model == RS_DISTORTION_MODIFIED_BROWN_CONRADY)
+                      {
+                          float r2 = dot(pixel, pixel);
+                          float f = 1.0 + u_color_coeffs[0] * r2 + u_color_coeffs[1] * r2 * r2 +
+                                    u_color_coeffs[4] * r2 * r2 * r2;
+                          pixel = pixel * f;
+                          float dx = pixel.x + 2.0 * u_color_coeffs[2] * pixel.x * pixel.y +
+                                     u_color_coeffs[3] * (r2 + 2.0 * pixel.x * pixel.x);
+                          float dy = pixel.y + 2.0 * u_color_coeffs[3] * pixel.x * pixel.y +
+                                     u_color_coeffs[2] * (r2 + 2.0 * pixel.y * pixel.y);
+                          pixel = vec2(dx, dy);
+                      }
+                      return pixel * u_color_focal_length + u_color_offset;
+                  }
+
+                  void main()
+                  {
+                      vec2 depth_pixel;
+                      // generate lattice pos; (0, 0) (1, 0) (2, 0) ... (w-1, h-1)
+                      depth_pixel.x = mod(float(gl_VertexID), u_depth_size.x);
+                      depth_pixel.y = clamp(floor(float(gl_VertexID) / u_depth_size.y), 0.0, u_depth_size.y);
```

Comment:
@ds-hwang Thanks for this. It is now online and running in Chrome browser: [https://github.com/01org/depthcamera-pointcloud-web-demo](https://github.com/01org/depthcamera-pointcloud-web-demo)